skrub/_squashing_scaler.py

```diff
@@ -0,0 +1,304 @@
+import numbers
+
+import numpy as np
+from sklearn.base import BaseEstimator, OneToOneFeatureMixin, TransformerMixin
+from sklearn.preprocessing import RobustScaler
+from sklearn.utils.validation import FLOAT_DTYPES, check_is_fitted
+
+from skrub._sklearn_compat import validate_data
+
+
+def _infinite_sign(X):
+    """Replace infinite values with NaN and return their sign."""
+    if (inf_sign := np.isinf(X)).any():
+        sign = np.sign(X)
+        X = np.where(inf_sign, np.nan, X)
+        # 0 when X is finite, 1 when X is +inf, -1 when X is -inf
+        inf_sign = inf_sign.astype(X.dtype) * sign
+
+    return X, inf_sign
+
+
+def _set_zeros(X, zero_cols):
+    """Set the finite values of the specified columns to zero."""
+    mask = np.isfinite(X)
+    mask[:, ~zero_cols] = False
+    X[mask] = 0.0
+    return X
+
+
+def _soft_clip(X, max_absolute_value, inf_sign):
+    """Apply a soft clipping to the data.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        The data to be clipped.
+    max_absolute_value : float, default=3.0
+        Maximum absolute value that the transformed data can take.
+    mask_inf : array-like, shape (n_samples, n_features)
+        A mask indicating the positions of infinite values in the input data.
+
+    Returns
+    -------
+    X_clipped : array-like, shape (n_samples, n_features)
+        The clipped version of the input.
+    """
+    X = X / np.sqrt(1 + (X / max_absolute_value) ** 2)
+    X = np.where(inf_sign == 1, max_absolute_value, X)
+    X = np.where(inf_sign == -1, -max_absolute_value, X)
+    return X
+
+
+class _MinMaxScaler(OneToOneFeatureMixin, TransformerMixin, BaseEstimator):
+    """A variation of scikit-learn MinMaxScaler.
+
+    A simple min-max scaler that centers the median to zero and scales
+    the data to the range [-2, 2].
+
+    scikit-learn MinMaxScaler computes the following::
+
+        X_std = (X - X.min(axis=0)) / (X.max(axis=0) - X.min(axis=0))
+        X_scaled = X_std * (max - min) + min
+
+    This scaler computes the following::
+
+        X_std = (X - median) / (X.max(axis=0) - X.min(axis=0) + eps)
+        X_scaled = X_std * (max - min) + min
+
+    where we set min = 0 and max = 2.
+    """
+
+    def fit(self, X, y=None):
+        del y
+        eps = np.finfo("float32").tiny
+        self.median_ = np.nanmedian(X, axis=0)
+        self.scale_ = 2 / (np.nanmax(X, axis=0) - np.nanmin(X, axis=0) + eps)
+        return self
+
+    def transform(self, X):
+        check_is_fitted(self, ["median_", "scale_"])
+        return self.scale_ * (X - self.median_)
+
+
+class SquashingScaler(OneToOneFeatureMixin, TransformerMixin, BaseEstimator):
+    r"""Perform robust centering and scaling of followed by soft clipping.
```

Comment:
Maybe the sentence is missing something? 

```suggestion
    r"""Perform robust centering and scaling followed by soft clipping.
```